source/nodejs/adaptivecards-templating/src/template-engine.ts

```diff
@@ -94,14 +97,24 @@ class TemplatizedString {
             }
         }
         else {
+            // If the templatized string has multiple parts, we want it to evaluate
+            // to a string. In that context, each part that evaluates to undefined
+            // gets replaced by the original expression by default or by a resource
+            // string provided by the application
             let s = "";
 
             for (let part of this._parts) {
                 if (typeof part === "string") {
                     s += part;
                 }
                 else {
-                    s += this.evalExpression(<Binding>part, context);
+                    let evaluatedPart = this.evalExpression(<Binding>part, context);
+
+                    if (evaluatedPart === undefined) {
+                        evaluatedPart = Shared.GlobalSettings.undefinedExpressionValueSubstitutionString ? Shared.GlobalSettings.undefinedExpressionValueSubstitutionString : (<Binding>part).expressionString;
```

Comment:
Minor change, but it seems this could be shortened to the following?
```
evaluatedPart =  Shared.GlobalSettings.undefinedExpressionValueSubstitutionString ?? (<Binding>part).expressionString;

Reply:
Indeed, changed.